csrc/id_model/indexing.cpp

```diff
@@ -210,8 +210,27 @@ class AllocationDomainSetup : private kir::IrVisitor {
     }
 
     if (use_set_allocation_domain) {
-      allocation_domains = tv->getAllocationDomain();
-      contiguity = tv->domain()->contiguity();
+      if (tv->getMemoryType() == MemoryType::Global) {
+        // For global memory tensors we always allocate the entire tensor
+        allocation_domains = tv->getAllocationDomain();
+        contiguity = tv->domain()->contiguity();
+      } else {
+        std::unordered_set<IterDomain*> exclude_ids;
+        for (auto i : c10::irange(tv->getComputeAtPosition())) {
```

Comment:
I suppose this is good enough inference for the matmul case, but in general, the CA position is with respect to the loop domain, whereas we are looking at the allocation domain here. 

Perhaps we should do something like:

```
auto inlined_domains = {tv->getLoopDomain().begin(), tv->getLoopDomain().begin() + tv->getComputeAtPosition()};
auto corresponding_allocation_domains = {tv->getAllocationDomain().begin(), tv->getAllocationDomain().begin() + tv->getComputeAtPosition()};
for (auto i: c10::irange(tv->getComputeAtPosition()) {
  NVF_ERROR(exact_graph.disjointValSets().strictAreMapped(inlined_domains[i], corresponding_allocation_domains[i]));
}
```

Reply:
Added a new commit: https://github.com/NVIDIA/Fuser/pull/2669/commits/77ec836748f65799adc49386448c970744cf319f